docusaurus/docs/React/custom-code-examples/video-integration-100ms.mdx

```diff
@@ -0,0 +1,881 @@
+
+Next, we’ll add the UI for initializing and ending calls. This should always be visible and a good place for that is the channel header. The SDK makes it easy to replace this and we can make use of this to inject a header that adds the functionality we need.
+
+Inside of the channel header we use the video context to take care of the logic. Let’s first create a folder for it in the `src` directory called `MyChannelHeader`. Inside of it, we’ll create two files, one called `MyChannelHeader.tsx` and one for styling called `MyChannelHeader.css`.
+
+First, take a look at the `tsx` file. We want it to be a component that shows the `channelName` so we need to get the current `channel` object from `useChannelStateContext()`. For that, we’ll use an import and then fetch it. Let’s first create a small dummy `<h2>` element that will show the channel name if available and `Unknown` otherwise:
+
+```tsx
+import { useChannelStateContext } from 'stream-chat-react';
+
+const MyChannelHeader = ({ channelName }: MyChannelHeaderProps) => {
+  const { channel } = useChannelStateContext();
+
+  return <h2>{channel?.data?.name ?? 'Unknown'}</h2>;
+};
+
+export default MyChannelHeader;
+```
+
+Next, we’ll use the `VideoContext` to import the necessary functions we need and create three helper functions:
+
+1. `onVideoButtonClick`: When the user clicks on the video button there should be different actions called depending on the current state of video from the context
+2. `callArea`: to show the correct button(s) or state indicators depending on the current `connectionState`
+3. `buttonText`: depending on the state of connection we will re-use the button to show different text
+
+Here is the code for the three functions, please add this inside of your `MyChannelHeader` component:
+
+```tsx
+const { connectionState, createCall, joinCall, leaveCall, endCall } = useVideoContext();
+
+const onVideoButtonClick = () => {
+  switch (connectionState) {
+    case ConnectionState.NoCall:
+      createCall();
+      break;
+    case ConnectionState.CallAvailable:
+      joinCall();
+      break;
+    case ConnectionState.InCall:
+      leaveCall();
+      break;
+  }
+};
+
+const callArea = (): ReactNode => {
+  switch (connectionState) {
+    case ConnectionState.NoCall:
+    case ConnectionState.CallAvailable:
+      return (
+        <button className='call-button start-call-button' onClick={onVideoButtonClick}>
+          {buttonText()}
+        </button>
+      );
+    case ConnectionState.Connecting:
+    case ConnectionState.Disconnecting:
+      return <p className='call-button'>{buttonText()}</p>;
+    case ConnectionState.InCall:
+      return (
+        <>
+          <button className='call-button leave-call-button' onClick={onVideoButtonClick}>
+            <p>{buttonText()}</p>
+          </button>
+          <button className='call-button end-call-button' onClick={endCall}>
+            <p>End call</p>
+          </button>
+        </>
+      );
+  }
+};
```

Comment:
I know this works and looks fine but by React standards it's an antipattern - meaning it's a function that returns React nodes so it as well could be a proper component.

Reply:
What would you consider the best way to do this then?

I feel like it's the best way to switch between the different states the enum can have and show UI accordingly. I know there's a way to do it [inside the DOM](https://stackoverflow.com/a/51432223) but that looks/feels even more ugly.